doc/Guide to Contributing to Sia.md

```diff
@@ -0,0 +1,253 @@
+
+To create and checkout a new branch:
+```bash
+# If you're not already in the right directory:
+$ cd $GOPATH/src/NebulousLabs/Sia
+# Make sure you're on branch master
+$ git checkout master
+# Create and checkout a new branch
+$ git checkout -b <branch>
+```
+Now write some code while the new branch is checked out.
+
+Only implement one logical change per branch. If you're working on several 
+things at once, make multiple branches. To switch between branches you're 
+working on, you have to stash the changes in the branch you're switching from 
+by running `git stash`, which tucks away all changes since the last 
+commit.
+
+```bash
+# Stash changes to current branch.
+$ git stash
+# Checkout other branch.
+$ git checkout <branch>
+...
+# Make changes
+...
+# Return to first branch:
+$ git checkout <branch 1>
+# View a list of stashes and their corresponding hashes.
+$ git stash list
+# Reapply changes from the stash you want to recover and remove that stash from.
+# the list
+$ git stash pop <hash>
+```
+
+To learn more about branching, see 
+[Using the Fork-and-Branch Git Workflow][branch] and 
+[Pro Git - Branches in a Nutshell][nutshell].
+For more on stashing, see [Pro Git - Stashing and Cleaning][stashing].
+  
+Be sure to follow the conventions detailed in 
+[docs/Developers.md][developers.md].  We will reject pull requests that do not 
+satisfy these best practices.
+
+Once you've finished making changes, stage and commit your changes then update 
+your fork on Github:
+
+```bash
+# Make sure the code is up to date with the original repo:
+$ git checkout master
+$ git pull origin master
+# Checkout branch with changes.
+$ git checkout <branch>
+$ git rebase master
+# Before every pull request, you should run `make test-long`
+# to test your code and fix formatting and style problems.
+# You need to be in the main directory if you're not already.
+$ cd $GOPATH/src/NebulousLabs/Sia
+$ make test-long
+# If all goes well, proceed to staging your changed files:
+$ git add <changed files>
+# Use `git status` to see what files have been staged.
+$ git status
+# Commit your changes. If you just run `commit`,  a text editor will pop up for 
+# you to enter a description of your changes.
+$ git commit -m "Add new tests for CommitSync method"
+# Push the changes to your fork on GitHub, which you should have set up as a 
+# remote already.
+$ git push <fork remote> <branch>
+```
+<a name="pull"/>
+### Submit your code for review
+Once you've tested your new code and pushed changes to your fork, navigate to 
+your fork at `https://github.com/<username>/Sia` in your browser.  
+Switch to the branch you've made changes on by selecting it from the list on 
+the upper left.  Then click 'New pull request' on the upper right.
+
+Once you have made the pull request, we will review your code.  We will reject 
+code that is unsafe, difficult to read, or otherwise violates the conventions 
+outlined in [docs/Developers.md][developers.md].
```

Comment:
Might be worth noting that most pull requests go through multiple revisions before being accepted. Would be pretty meta if you included a screenshot of this comment in the guide as an example of code review.